e2e/update_pnpm_lock_with_import/test.sh

```diff
@@ -27,18 +38,30 @@ _sedi 's#"@types/node": "18.11.18"#"@types/node": "16"#' package.json
 
 export ASPECT_RULES_JS_FROZEN_PNPM_LOCK=1
 
-if bazel test //...; then
-  echo "ERROR: expected 'ASPECT_RULES_JS_FROZEN_PNPM_LOCK=1 bazel test //...' to fail"
+if bazel test $BZLMOD_FLAG //...; then
+  echo "ERROR: expected 'ASPECT_RULES_JS_FROZEN_PNPM_LOCK=1 bazel test $BZLMOD_FLAG //...' to fail"
   exit 1
 fi
 
 ASPECT_RULES_JS_FROZEN_PNPM_LOCK=
 
-if ! bazel sync --only=npm; then
-  echo "ERROR: expected 'bazel sync --only=npm' to pass"
-  exit 1
+if [ ! $BZLMOD_FLAG ]; then
+  print_step "It should update the lockfile after a bazel sync with ASPECT_RULES_JS_FROZEN_PNPM_LOCK unset"
+  if ! bazel sync --only=npm; then
+    echo "ERROR: expected 'bazel sync --only=npm' to pass"
+    exit 1
+  fi
+else
+  print_step "It should update the lockfile after a failed bazel build with ASPECT_RULES_JS_FROZEN_PNPM_LOCK unset"
+  # bazel sync isn't load bearing under bzlmod.
```

Comment:
hmmm. the instructions in
```
def _fail_if_frozen_pnpm_lock(rctx, state):
    if RULES_JS_FROZEN_PNPM_LOCK_ENV in rctx.os.environ.keys() and rctx.os.environ[RULES_JS_FROZEN_PNPM_LOCK_ENV]:
        fail("""

ERROR: `{action_cache}` is out of date. `{pnpm_lock}` may require an update. To update run,

           bazel sync --only={rctx_name}

""".format(
``` 
are wrong then under bzlmod

Reply:
I also left a `use: bazel sync --only npm` reference in the docstring of `use_home_npmrc` that needs updating to handle the bzlmod case
```
        use_home_npmrc: Use the `$HOME/.npmrc` file (or `$USERPROFILE/.npmrc` when on Windows) if it exists.
            ...
            WARNING: The repository rule will not be invalidated by changes to the home `.npmrc` file since there
            is no way to specify this file as an input to the repository rule. If changes are made to the home
            `.npmrc` you can force the repository rule to re-run and pick up the changes by running:
            `bazel sync --only={name}` where `name` is the name of the `npm_translate_lock` you want to re-run.
```